### PR TITLE
improvement: pre-release testing

### PR DESCRIPTION
**Pull request by dsinghvi**

### CODE REVIEW

**Comment by dsinghvi** [comment]
Just replied in Slack -- plan is to just have this run right before you tag just so ik all the generator versions in `fern add` work fine

**Comment by dsinghvi** [comment]
Do you know how i can make the `fern generate` output less "interactive" and more just update based

**Comment by zachkirsch** [comment]
Nothing cheap - right now we just wipe and repaint every x milliseconds. It’s stateless, it just paints the latest reply from the BE. To only log updated we’ll need to change it to be stateful so it can tell what changed between polls

File: .circleci/config.yml [hunk 1]
```diff
@@ -57,6 +57,22 @@ jobs:
...
+      - run: |
+          npm install -g fern-api@next
+          mkdir pre-release-test
+          cd pre-release-test
+          fern init --organization fern
+          fern add java
+          fern add typescript
+          fern add postman
+          fern generate
+          rm -rf pre-release-test
```
Comment (zachkirsch) [inline .circleci/config.yml:74]: i think there should be a `cd ..` first. and maybe `rm -r`, the `-f` is probably swallowing that error

File: .circleci/config.yml [hunk 2]
```diff
@@ -57,6 +57,22 @@ jobs:
           name: Ensure no changes to git-tracked files
           command: git --no-pager diff
 
+  pre-release-test:
+    machine:
+      image: ubuntu-2004:current
```
Comment (zachkirsch) [inline .circleci/config.yml:62]: no idea if machine is more expensive, but if it is we don't need docker for this one

File: .circleci/config.yml [hunk 3]
```diff
@@ -111,6 +127,10 @@ workflows:
               only: /.*/
           context:
             - auth0-dev
+      - pre-release-test:
+          filters:
+            tags:
+              only: /.*/
```
Comment (zachkirsch) [inline .circleci/config.yml:133]: youre prob tracking but rn this will run on all commits

Comment (dsinghvi) [reply .circleci/config.yml:138]: yup will restrict to main branch

Comment (zachkirsch) [reply .circleci/config.yml:141]: do we want it to run pre publishing snapshots or just releases

File: .circleci/config.yml (hunk 1 above)
Comment (zachkirsch) [reply .circleci/config.yml]: @dsinghvi

Comment (dsinghvi) [reply .circleci/config.yml]: ah okay

File: .circleci/config.yml (hunk 3 above)
Comment (dsinghvi) [reply .circleci/config.yml:151]: just releases

Comment (dsinghvi) [reply .circleci/config.yml:151]: there's an if case that terminates early if there is no git tag

File: .circleci/config.yml [hunk 4]
```diff
@@ -57,6 +57,27 @@ jobs:
...
+      - image: cimg/node:current
+    steps:
+      - checkout
+      - node/install-packages:
+          pkg-manager: yarn-berry
+      - run:
+          name: ETE Test
+          command: |
+            yarn workspace @fern-api/cli dist
+            cli_path = yarn workspace @fern-api/cli exec pwd
```
[the remark this answers is not in the document]
Comment (dsinghvi) [reply .circleci/config.yml]: thanks

**Pull request merged**